### DISCUSSION TITLE
Quantifier Elimination with non Quantified Top Level Formula

### DISCUSSION THREAD

**Discussion by baierd** [cvc5/cvc5#8593] [Q&A] [answered]
Hello CVC5 team,
is it possible to use quantifier elimination on formulas that are not quantified on the upmost level, i.e. `(and (forall ...) (exists ...))`?
By default the option `finite-model-find` is turned on in our project. We use the Java API of CVC5. Are there options that enable the desired behavior?

Best,
Daniel

**Comment by ajreynol** [comment, marked as answer]
At the moment,  the quantifier elimination method `getQuantifierElimination`, and the smt2 command `get-qe` require that the provided formula is a top-level quantified formula. 

It wouldn't be very difficult to lift this restriction. Would this be helpful to you?

The option `finite-model-find` shouldn't interfere with quantifier elimination, although it won't be used for solving the quantifier elimination problem.
